### PR TITLE
feat(firestore): Distance result field and distance threshold in vector search

### PR DESCRIPTION
**Pull request by bhshkh**
## Checklist
- [x] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [x] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [x] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [x] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Comment by bhshkh** [comment]
To be merged after:
https://github.com/googleapis/google-cloud-go/pull/10802 
https://github.com/GoogleCloudPlatform/golang-samples/pull/4358
https://github.com/googleapis/google-cloud-go/pull/10597

File: firestore/vector_search_distance_threshold.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+	client, err := firestore.NewClient(ctx, projectID)
+	if err != nil {
+		return fmt.Errorf("firestore.NewClient: %w", err)
+	}
+	defer client.Close()
+
+	collection := client.Collection("coffee-beans")
+
+	// Requires a vector index
```
Comment (telpirion) [inline firestore/vector_search_distance_threshold.go:37]: nit: provide some more context around what a "vector index" is. Maybe provide a link to documentation on cloud.google.com?

Here and following.

File: firestore/vector_search_result_field.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+	client, err := firestore.NewClient(ctx, projectID)
+	if err != nil {
+		return fmt.Errorf("firestore.NewClient: %w", err)
+	}
+	defer client.Close()
+
+	collection := client.Collection("coffee-beans")
+
+	// Requires a vector index
```
Comment (telpirion) [inline firestore/vector_search_result_field.go:37]: See previous.

File: firestore/vector_search_result_field_masked.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+	client, err := firestore.NewClient(ctx, projectID)
+	if err != nil {
+		return fmt.Errorf("firestore.NewClient: %w", err)
+	}
+	defer client.Close()
+
+	collection := client.Collection("coffee-beans")
+
+	// Requires a vector index
```
Comment (telpirion) [inline firestore/vector_search_result_field_masked.go:37]: See previous.

**Review by telpirion — APPROVED** [review]
Only small nits--address as you see fit. I'll merge this PR after your next comment. Thanks!

File: firestore/vector_search_distance_threshold.go (hunk 1 above)
Comment (bhshkh) [reply firestore/vector_search_distance_threshold.go:37]: Added link to documentation

**Pull request merged**